src/java.base/share/classes/java/lang/invoke/LambdaMetafactory.java

```diff
@@ -425,80 +438,122 @@ public static CallSite metafactory(MethodHandles.Lookup caller,
      *               full privilege access}.
      *               When used with {@code invokedynamic}, this is stacked
      *               automatically by the VM.
-     * @param invokedName The name of the method to implement.  When used with
-     *                    {@code invokedynamic}, this is provided by the
-     *                    {@code NameAndType} of the {@code InvokeDynamic}
-     *                    structure and is stacked automatically by the VM.
-     * @param invokedType The expected signature of the {@code CallSite}.  The
+     * @param interfaceMethodName The name of the method to implement.  When used with
+     *                            {@code invokedynamic}, this is provided by the
+     *                            {@code NameAndType} of the {@code InvokeDynamic}
+     *                            structure and is stacked automatically by the VM.
+     * @param factoryType The expected signature of the {@code CallSite}.  The
      *                    parameter types represent the types of capture variables;
      *                    the return type is the interface to implement.   When
      *                    used with {@code invokedynamic}, this is provided by
      *                    the {@code NameAndType} of the {@code InvokeDynamic}
      *                    structure and is stacked automatically by the VM.
-     *                    In the event that the implementation method is an
-     *                    instance method and this signature has any parameters,
-     *                    the first parameter in the invocation signature must
-     *                    correspond to the receiver.
-     * @param  args       An {@code Object[]} array containing the required
-     *                    arguments {@code samMethodType}, {@code implMethod},
-     *                    {@code instantiatedMethodType}, {@code flags}, and any
-     *                    optional arguments, as described
-     *                    {@link #altMetafactory(MethodHandles.Lookup, String, MethodType, Object...)} above}
+     * @param  args An array of {@code Object} containing the required
+     *              arguments {@code interfaceMethodType}, {@code implementation},
+     *              {@code dynamicMethodType}, {@code flags}, and any
+     *              optional arguments, as described above
      * @return a CallSite whose target can be used to perform capture, generating
-     *         instances of the interface named by {@code invokedType}
-     * @throws LambdaConversionException If any of the linkage invariants
-     *                                   described {@link LambdaMetafactory above}
-     *                                   are violated, or the lookup context
-     *                                   does not have private access privileges.
+     *         instances of the interface named by {@code factoryType}
+     * @throws LambdaConversionException If {@code caller} does not have private
```

Comment:
The lookup access has been extended since 14 to include `MODULE` and `ORIGINAL` access.  
`Lookup::hasFullPrivilegeAccess` returns true if the lookup has `PRIVATE` and `MODULE` which means that this lookup is not teleported from another module via `Lookup::in` and `MethodHandles::privateLookupIn`.

What privilege do you expect the `caller` lookup have?  I believe full privilege access is the appropriate privilege.  The `ORIGINAL` access is stricter as the lookup must be created from the original lookup class.

[1] shows what access a `Lookup` has when being produced via different APIs.

[1] https://download.java.net/java/early_access/jdk17/docs/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#access-modes

Reply:
Thanks, yes I think `hasFullPrivilegeAccess` is what we want. I updated the javadoc, along with the actual check. Thanks for catching!